mzLib/Proteomics/RetentionTimePrediction/Chronologer/ChronologerEstimator.cs

```diff
@@ -0,0 +1,172 @@
+                            tensor[0][0] = 39; 
+                        else
+                        {
+                            tensor[0][0] = 38; 
+                        }
+                        nTerminalMod = false; //next iteration is not a mod
+                        continue;
+                    }
+
+                    if (!subString.Contains(" "))
+                    {
+                        //without mods
+                        for (int i = 0; i < subString.Length; i++)
+                        {
+                            tensor[0][tensorCounter] = ChronologerDictionary[(subString[i], "")];
+                            tensorCounter++;
+                        }
+
+                        //save target aa for next loop
+                        modID = subString[subString.Length - 1];
+                    }
+
+                    nTerminalMod = true;
+                }
+
+                tensor[0][tensorCounter] = 44; //C-terminus
+
+                return tensor;
+            }
+
+            return null;
+
+        }
+
+        private static readonly Dictionary<(char, string), int> ChronologerDictionary = new()
+            {
+                { ('A', ""), 1 }, //'Alanine
+                { ('C', ""), 2 }, //'Cysteine
+                { ('D', ""), 3 }, //'Aspartate
+                { ('E', ""), 4 }, //'Glutamate
+                { ('F', ""), 5 }, //'Phenylalanine
+                { ('G', ""), 6 }, //'Glycine
+                { ('H', ""), 7 }, //'Histidine
+                { ('I', ""), 8 }, //'Isoleucine
+                { ('K', ""), 9 }, //'Lysine
+                { ('L', ""), 10 }, //'Leucine
+                { ('M', ""), 11 }, //'Methionine
+                { ('N', ""), 12 }, //'Asparagine
+                { ('P', ""), 13 }, //'Proline
+                { ('Q', ""), 14 }, //'Glutamine
+                { ('R', ""), 15 }, //'Argenine
+                { ('S', ""), 16 }, //'Serine
+                { ('T', ""), 17 }, //'Threonine
+                { ('V', ""), 18 }, //'Valine
+                { ('W', ""), 19 }, //'Tryptophane
+                { ('Y', ""), 20 }, //'Tyrosine
+                { ('C', "Carbamidomethyl on C"), 21 }, //'Carbamidomethyl
+                { ('M', "Oxidation on M"), 22 }, //'Oxidized
+                { ('E', "Glu to PyroGlu"), 24 }, //'Pyroglutamate
+                { ('S', "Phosphorylation on S"), 25 }, //'Phosphoserine
+                { ('T', "Phosphorylation on T"), 26 }, //'Phosphothreonine
+                { ('Y', "Phosphorylation on Y"), 27 }, //'Phosphotyrosine
+                { ('K', "Accetylation on K"), 28 }, //'Acetylated
+                { ('K', "Succinylation on K"), 29 }, //'Succinylated
+                { ('K', "Ubiquitination on K"), 30 }, //'Ubiquitinated
+                { ('K', "Methylation on K"), 31 }, //'Monomethyl
+                { ('K', "Dimethylation on K"), 32 }, //'Dimethyl
+                { ('K', "Trimethylation on K"), 33 }, //'Trimethyl
```

Comment:
please update the comments above to include these mods and any conversion info needed